File: sprint7/По алгоритмам/lesson10.3/lesson10code3.kt
val dp = Array(n + 1) { Array(m + 1) { 0 } }
for (i in 1..n) {
    var maxVal = 0
    for (j in 1..m) {
        dp[i][j] = dp[i - 1][j]
        if (A[i - 1] == B[j - 1]) {
            dp[i][j] = maxOf(dp[i][j], maxVal + 1)
        }
        if (B[j - 1] < A[i - 1]) {
            maxVal = maxOf(maxVal, dp[i - 1][j])
        }
    }
}

File: sprint7/By lessons/lesson10code3/lesson10code3.js
const dp = Array.from({ length: n + 1 }, () => Array(m + 1).fill(0));
for (let i = 1; i <= n; i++) {
  let max_val = 0;
  for (let j = 1; j <= m; j++) {
    dp[i][j] = dp[i - 1][j];
    if (A[i - 1] === B[j - 1]) {
      dp[i][j] = Math.max(dp[i][j], max_val + 1);
    }
    if (B[j - 1] < A[i - 1]) {
      max_val = Math.max(max_val, dp[i - 1][j]);
    }
  }
}
